tests/hmm/hmumu_utils.py

```diff
@@ -1914,6 +1915,33 @@ def get_new_btag_weights_shape(jets, evaluator, era, scalars, pt_cut):
         #print(p_jetWt_down[i][mask_pt_bounds[i]],jets.pt[mask_pt_bounds[i]],jets.eta[mask_pt_bounds[i]],jets.btagDeepB[mask_pt_bounds[i]])
     return eventweight_btag , eventweight_btag_up, eventweight_btag_down
 
+#qgl reweighting
+def get_qglWeights(jets, jet_attrs, ret_mu, func, dataset_name):
+    qglw = NUMPY_LIB.ones(jets.numevents())
+    qglw_up = NUMPY_LIB.ones(jets.numevents())
+    psgen = NUMPY_LIB.ones(jets.numevents())
+    if "pythia" in dataset_name:
+        psgen = NUMPY_LIB.zeros(jets.numevents())
+    leading_jet = jets.select_nth(
+                0, ret_mu["selected_events"], None,
+                attributes=jet_attrs)
+    subleading_jet = jets.select_nth(
+                1, ret_mu["selected_events"], None,
+                attributes=jet_attrs)
+    qglj1w = NUMPY_LIB.array(func.qglJetWeight(
+                    NUMPY_LIB.asnumpy(leading_jet["partonFlavour"]),
+                    NUMPY_LIB.asnumpy(leading_jet["pt"]),
+                    NUMPY_LIB.asnumpy(leading_jet["eta"]),
+                    NUMPY_LIB.asnumpy(leading_jet["qgl"]), psgen))
+    qglj2w = NUMPY_LIB.array(func.qglJetWeight(
+                    NUMPY_LIB.asnumpy(subleading_jet["partonFlavour"]),
+                    NUMPY_LIB.asnumpy(subleading_jet["pt"]),
+                    NUMPY_LIB.asnumpy(subleading_jet["eta"]),
+                    NUMPY_LIB.asnumpy(subleading_jet["qgl"]), psgen))
+    qglw = NUMPY_LIB.multiply(qglj1w,qglj2w)
```

Comment:
here `qglj1w* qglj2w` would also work, perhaps simpler to read :)

Reply:
Sure, done.